### PR TITLE
[ONNX] Remove unused patching methods

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83008
* #83007
* __->__ #83006

### Description

Remove unused patching methods:

- `torch._C.Graph.constant`
- unpatch `torch._C.Node.__getitem__` and move the helper function to `symbolic_helper`

Add typing annotations

### Issue

#76254

### Testing

Unit tested

### CODE REVIEW

File: torch/onnx/symbolic_helper.py [hunk 1]
```diff
@@ -131,6 +131,12 @@ def _parse_arg(value, desc, arg_name=None, node_name=None):
         )
 
 
+def _node_get(node: _C.Node, key: str):
```
Comment (BowenBao) [inline torch/onnx/symbolic_helper.py:134]: Should we mention `attr` in the function name?

Comment (justinchuby) [reply torch/onnx/symbolic_helper.py:134]: I think we can opt for simplicity in this case, since I don't foresee getting other things from the node. If it becomes necessary that we distinguish between different get methods, we can rename the function or design a more elaborate mechanism for working with nodes.

File: torch/onnx/utils.py [hunk 2]
```diff
@@ -1703,38 +1703,38 @@ def _run_symbolic_function(
...
     try:
         symbolic_registry.register_version("", opset_version)
 
         # Caffe2-specific: Quantized op symbolics are registered for opset 9 only.
         if symbolic_helper.is_caffe2_aten_fallback() and opset_version == 9:
-
             symbolic_caffe2.register_quantized_ops("caffe2", opset_version)
 
-        if ns == "aten":
+        if namespace == "aten":
```
Comment (BowenBao) [inline torch/onnx/utils.py:1724]: nit: Some of these are not related to "Remove unused patching methods"...

**Review by BowenBao — APPROVED** [review]

File: torch/onnx/utils.py (hunk 2 above)
Comment (justinchuby) [reply torch/onnx/utils.py:1724]: I will break them out

File: torch/onnx/symbolic_helper.py (hunk 1 above)
Comment (justinchuby) [reply torch/onnx/symbolic_helper.py:134]: But this also raises a good point on how we can supply a set of helpers to users who what to create their own symbolics. if one needs to obtain a value in a node, all they can do now is replicate this function and use that. Since we are usually calling private helpers, users may follow our usage and inevitably use those private helpers too

Comment (BowenBao) [reply torch/onnx/symbolic_helper.py:134]: sgtm

File: torch/onnx/utils.py (hunk 2 above)
Comment (justinchuby) [reply torch/onnx/utils.py:1724]: Actually I will keep them in if you are ok with it, since this function is touched by the removal (https://github.com/pytorch/pytorch/pull/83006/files/b305518f3ed8d57b78dcfcaeb861c6c53c2a5df2#diff-849a5778e2dcf7f36587967273cee0bf20642e35bf4c79405111ea3417c3fb3cR1748-R1751 etc) and a mild clean up doesn't seem too noisy?

Comment (justinchuby) [reply torch/onnx/utils.py:1724]: Mark as resolved for this one. I will break changes like this out in the future. Thanks!
